### PR TITLE
Added support for full domain urls for cached images

### PR DESCRIPTION
**Pull request by kaidoj**
I needed to add support for cookieless domain, static files domain. Now you have new attribute in config cache_url where you can add external url like http://static.example.com. If this is set then  that url is used. I think i am not the only one who needs it.

### CODE REVIEW

**Comment by ojhaujjwal** [comment]
Can you add tests to show me that it works?

File: src/Service/CacheManager.php [hunk 1]
```diff
@@ -65,13 +65,13 @@ public function getCachePath($relativeName, $filter, $formatOrImage = null)
     /**
      * {@inheritDoc}
      */
-    public function createCache($relativeName, $filter, ImageInterface $image, $formatOrImage = null, array $saveOptions = [])
+    public function createCache($relativeName, $filter, ImageInterface $image, $formatOrImage = null)
```
Comment (ojhaujjwal) [inline src/Service/CacheManager.php]: Why did you remove this the last argument? Please add it back.

File: src/Service/CacheManager.php [hunk 2]
```diff
@@ -96,4 +96,16 @@ public function isCachingEnabled($filter, array $filterOptions)
 
         return $this->cacheOptions->getEnableCache();
     }
+    
+    /**
+     * {@inheritDoc}
+     */
+    public function useCacheUrl()
```
Comment (ojhaujjwal) [inline src/Service/CacheManager.php]: Can you add this method to CacheManagerInterface with the appropriate docblocks?

File: src/Service/CacheManager.php [hunk 3]
```diff
@@ -96,4 +96,16 @@ public function isCachingEnabled($filter, array $filterOptions)
...
+    {
+        if($this->cacheOptions->getCacheUrl()) {
+            $this->cacheOptions->setCachePath('');
+            return $this->cacheOptions->getCacheUrl();
+        }
+        return null;
```
Comment (ojhaujjwal) [inline src/Service/CacheManager.php]: `return false`?

File: src/Service/CacheManager.php [hunk 4]
```diff
@@ -96,4 +96,16 @@ public function isCachingEnabled($filter, array $filterOptions)
...
         return $this->cacheOptions->getEnableCache();
     }
+    
+    /**
+     * {@inheritDoc}
+     */
+    public function useCacheUrl()
+    {
+        if($this->cacheOptions->getCacheUrl()) {
+            $this->cacheOptions->setCachePath('');
```
Comment (ojhaujjwal) [inline src/Service/CacheManager.php]: I don't think overriding any configuration is a good idea!

File: src/View/Helper/ImgUrl.php [hunk 5]
```diff
@@ -59,8 +60,12 @@ public function __invoke($relativeName, $filter)
             $format = $binary->getFormat() ?: 'png';
         }
         if ($this->cacheManager->isCachingEnabled($filter, $filterOptions) && $this->cacheManager->cacheExists($relativeName, $filter, $format)) {
+    
+            if($this->cacheManager->useCacheUrl()) {
+                return $this->cacheManager->useCacheUrl() . $this->cacheManager->getCacheUrl($relativeName, $filter, $format);
```
Comment (ojhaujjwal) [inline src/View/Helper/ImgUrl.php]: How about adding this logic on `HtImgModule\Service\CacheManagerInterface::getCacheUrl` method instead?

**Comment by ojhaujjwal** [comment]
I like the overall idea. Thanks!

**Comment by icansurvive** [comment]
Changed the code more elegant. Im not really a test writer and i dont really know how. I can do it from scratch but not when you have errors in there when i run phpunit before i even have done something. Im using it in my project and it is working fine. Feel free to write test and show me how its done so i can maby learn something about it.

**Comment by ojhaujjwal** [comment]
@icansurvive Please checkout the official documentation of phpunit. It is very easy. If you are good enough to send the PR, you are definitely good enough to write tests. I will tell you an interesting thing. The first time I ever used phpunit was in this module. Since that, there was no turning back.

You can just see the current tests to learn how to write tests.

File: src/Service/CacheManager.php [hunk 6]
```diff
@@ -96,4 +104,15 @@ public function isCachingEnabled($filter, array $filterOptions)
 
         return $this->cacheOptions->getEnableCache();
     }
+    
+    /**
+     * {@inheritDoc}
+     */
+    public function useCacheUrl()
+    {
+        if($this->cacheOptions->getCacheUrl()) {
```
Comment (ojhaujjwal) [inline src/Service/CacheManager.php]: `return $this->cacheOptions->getCacheUrl() ? true: false;`

**Comment by ojhaujjwal** [comment]
Not sure how the tests does not run, the [builds](https://travis-ci.org/hrevert/HtImgModule) did not show any problem with tests.

**Comment by icansurvive** [comment]
@ojhaujjwal dunno i went console to tests dir and run command phpunit and i got PHP Fatal error:  Class 'Phine\Test\Property' not found etc.

**Comment by icansurvive** [comment]
Okay. Did test and and everything passed. No failures.I am not sure it is the right way. Let me know is it ok.
